pkg/db/src/implementations/version-1.2/utils.ts

```diff
@@ -68,10 +68,12 @@ export const addAvailableWarehouses = (
 ): Iterable<VolumeStockClient> => {
 	return map(entries, (e) => {
 		const availableWarehouses: NavMap<{ quantity: number }> = new Map(
-			map(stock.isbn(e.isbn), ([[, warehouseId], { quantity = 0 }]) => [
-				warehouseId,
-				{ displayName: warehouses.get(warehouseId)?.displayName || "not-found", quantity }
-			])
+			wrapIter(stock.isbn(e.isbn))
+				.filter(([[, warehouseId]]) => warehouses.get(warehouseId) != null)
+				.map(([[, warehouseId], { quantity = 0 }]) => [
+					warehouseId,
+					{ displayName: warehouses.get(warehouseId)?.displayName || "not-found", quantity }
+				])
```

Comment:
On the plus side, this is a quick and easy solution, making the app work as we'd expect it to. Additionally, it displays a good understanding of the generators library we're using for iterable operations.

However, this is a patch-me-up solution where we're "treating the symptoms, not the cause". We're not removing the warehouse stock from existence, but rather adding more code to block the data that we no-longer want there from being shown up and, as such, leads to unnecessary entropy (both in code and in the data).

Can we move this fix upstream (so that the stock passed here doesn't include the non-existing warehouses to begin with)